### ISSUE TITLE
Unused param not reported when missing description

### ISSUE THREAD

**Issue by aaime** [checkstyle/checkstyle#6743]
```
/var/tmp $ javac EmptyMethodParams.java
# no output produced, compiles properly

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Check Configuration 1.3//EN" 
"https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocMethod">
      <property name="allowMissingParamTags" value="true"/>
      <property name="allowMissingThrowsTags" value="true"/>
      <property name="allowMissingReturnTag" value="true"/>
      <property name="allowMissingJavadoc" value="true"/>
      <property name="allowMissingPropertyJavadoc" value="true"/>
      <property name="suppressLoadErrors" value="true"/>
    </module>
  </module>
</module>


/var/tmp $ cat EmptyMethodParams.java
import java.io.IOException;

public class EmptyMethodParams {

    /**
     * A method with an undocumented param and a missing param.
     * Checkstyle should at least
     * missingParam, ideally would be nice to have a flag to report
     * also undocumentedParam
     *
     * @param param1 some text
     * @param unexpectedParam
     * @return
     */
    void testEmpty(Object param1) {
            
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.20-all.jar \
     -c config.xml EmptyMethodParams.java
Starting audit...
Audit done.
```

---------------

The `unexpectedParam` is not reported as being mismatched, however, adding even a single character of description to it, results in the output I was expecting:

``
Starting audit...
[ERROR] /home/aaime/devel/checkstyle-empty-report/src/EmptyMethodParams.java:12:8: Unused @param tag for 'unexpectedParam'. [JavadocMethod]
Audit done.
``

FYI: #6746

**Comment by romani** [comment]
> Ideally, it would be nice if also a flag to report undocumentedParam, but this may have to go in a separate improvement request.

yes, should be separate.

**Comment by aaime** [comment]
Gave it a shot at https://github.com/checkstyle/checkstyle/issues/6746

**Comment by romani** [comment]
fix is merged.
